src/GitHub.App/Services/GitClient.cs

```diff
@@ -394,6 +401,20 @@ public Task<bool> IsHeadPushed(IRepository repo)
             });
         }
 
+        Task Fetch(IRepository repo, UriString cloneUrl, params string[] refspecs)
+        {
+            var tempRemoteName = $"{cloneUrl.Owner}-{Guid.NewGuid()}";
```

Comment:
This should only create temporary remotes when necessary (when cloneUrl != origin).